### PR TITLE
Add contact folder setting for hiding contact images

### PR DESCRIPTION
**Pull request by Nachtalb**
Add contactfolder setting for hiding contact images in listing view.

### CODE REVIEW

File: ftw/contacts/tests/test_hide_contact_images.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        contactfolder = create(
+            Builder('contact folder').titled(u'Contact folder').having(
+                hide_contacts_image=False))
+
+        contact = create(Builder('contact')
+                         .with_maximal_info(u'Chuck', u'Norris')
+                         .within(contactfolder))
+
+        browser.login().visit(contact, view='contact_summary')
+        import pdb; pdb.set_trace()
```
Comment (mbaechtold) [inline ftw/contacts/tests/test_hide_contact_images.py]: ✂️

File: ftw/contacts/tests/test_hide_contact_images.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    def setUp(self):
+        self.portal = self.layer['portal']
+        setRoles(self.portal, TEST_USER_ID, ['Manager'])
+
+    @browsing
+    def test_contact_images_are_hidden_when_hide_images_checkbox_is_active(self, browser):
+        contactfolder = create(
+            Builder('contact folder').titled(u'Contact folder').having(
+                hide_contacts_image=True))
```
Comment (mbaechtold) [inline ftw/contacts/tests/test_hide_contact_images.py]: > When applying a PEP8 guideline would make the code less readable, this particular guideline should be ignored.

Source: https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds

My question: does this line break make the more or less readable?

File: ftw/contacts/tests/test_hide_contact_images.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        browser.login().visit(contact, view='contact_summary')
+
+        self.assertEqual(1, len(browser.css('div.contactSummary')))
+        self.assertEqual(0, len(browser.css('a.contactImage')))
+
+    @browsing
+    def test_contact_images_are_shown_when_hide_images_checkbox_is_deactivated(self, browser):
+        contactfolder = create(
+            Builder('contact folder').titled(u'Contact folder').having(
+                hide_contacts_image=False))
```
Comment (mbaechtold) [inline ftw/contacts/tests/test_hide_contact_images.py]: > When applying a PEP8 guideline would make the code less readable, this particular guideline should be ignored.

Source: https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds

My question: does this line break make the more or less readable?

File: ftw/contacts/contents/contactfolder.py [hunk 4]
```diff
@@ -2,12 +2,20 @@
...
+from zope import schema
+from ftw.contacts import _
 
 
 class IContactFolderSchema(model.Schema):
     """A contactfolder type schema interface
     """
 
+    hide_contacts_image = schema.Bool(
+        title=_(u'contactdirectory_hide_contacts_image', default=u'Hide images of contacts'),
```
Comment (mbaechtold) [inline ftw/contacts/contents/contactfolder.py]: According to our guidelines we use a message id of the form `label_<name_of_the_file>` for the title.

File: ftw/contacts/contents/contactfolder.py [hunk 5]
```diff
@@ -2,12 +2,20 @@
...
+        default=False,
+        description=_(u'If activated, the images of the contacts are not shown.')
```
Comment (mbaechtold) [inline ftw/contacts/contents/contactfolder.py]: According to our guidelines we use a message id of the form `help_<name_of_the_file>` for the description.

File: ftw/contacts/contents/contactfolder.py [hunk 6]
```diff
@@ -2,12 +2,20 @@
 from plone.dexterity.content import Container
 from plone.supermodel import model
 from zope.interface import implements
+from zope import schema
+from ftw.contacts import _
```
Comment (mbaechtold) [inline ftw/contacts/contents/contactfolder.py]: We use to sort the imports alphabetically.

File: ftw/contacts/locales/de/LC_MESSAGES/ftw.contacts.po [hunk 7]
```diff
@@ -38,6 +38,10 @@ msgstr "vCard herunterladen"
 msgid "ID of the LDAP PAS plugin to use for contact synchronisation"
 msgstr ""
 
+#: ./ftw/contacts/contents/contactfolder.py:16
+msgid "If activated, the images of the contacts are not shown."
+msgstr "Wenn aktivert, werden die Bileder der Kontakte nicht angezeigt."
```
Comment (mbaechtold) [inline ftw/contacts/locales/de/LC_MESSAGES/ftw.contacts.po]:
```diff
-Bileder
+Bilder
```

**Comment by Nachtalb** [comment]
@mbaechtold requested changes were made

File: ftw/contacts/tests/test_hide_contact_images.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
+from ftw.builder import Builder
+from ftw.builder import create
+from ftw.contacts.testing import FTW_CONTACTS_FUNCTIONAL_TESTING
+from ftw.testbrowser import browsing
+from plone.app.testing import TEST_USER_ID
+from plone.app.testing import setRoles
+from unittest2 import TestCase
+
+
+class TestHideContactImages(TestCase):
```
Comment (mbaechtold) [inline ftw/contacts/tests/test_hide_contact_images.py]: You could inherit from `ftw.contacts.tests.FunctionalTestCase`.

File: ftw/contacts/tests/test_hide_contact_images.py [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+    layer = FTW_CONTACTS_FUNCTIONAL_TESTING
```
Comment (mbaechtold) [inline ftw/contacts/tests/test_hide_contact_images.py]: ✂️ if you inherit from `ftw.contacts.tests.FunctionalTestCase`.

File: ftw/contacts/tests/test_hide_contact_images.py [hunk 10]
```diff
@@ -0,0 +1,45 @@
+from ftw.builder import Builder
+from ftw.builder import create
+from ftw.contacts.testing import FTW_CONTACTS_FUNCTIONAL_TESTING
+from ftw.testbrowser import browsing
+from plone.app.testing import TEST_USER_ID
+from plone.app.testing import setRoles
+from unittest2 import TestCase
```
Comment (mbaechtold) [inline ftw/contacts/tests/test_hide_contact_images.py]: ✂️ if you inherit from `ftw.contacts.tests.FunctionalTestCase`.

File: ftw/contacts/tests/test_hide_contact_images.py [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+
+class TestHideContactImages(TestCase):
+
+    layer = FTW_CONTACTS_FUNCTIONAL_TESTING
+
+    def setUp(self):
+        self.portal = self.layer['portal']
```
Comment (mbaechtold) [inline ftw/contacts/tests/test_hide_contact_images.py]: ✂️ if you inherit from `ftw.contacts.tests.FunctionalTestCase`.

File: ftw/contacts/tests/test_hide_contact_images.py [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+        setRoles(self.portal, TEST_USER_ID, ['Manager'])
```
Comment (mbaechtold) [inline ftw/contacts/tests/test_hide_contact_images.py]: Use `self.grant('Manager') if you inherit from `ftw.contacts.tests.FunctionalTestCase`.

**Comment by Nachtalb** [comment]
@mbaechtold please review

File: ftw/contacts/tests/test_hide_contact_images.py [hunk 13]
```diff
@@ -0,0 +1,40 @@
+from ftw.builder import Builder
+from ftw.builder import create
+from ftw.testbrowser import browsing
+from ftw.contacts.tests import FunctionalTestCase
```
Comment (mbaechtold) [inline ftw/contacts/tests/test_hide_contact_images.py]: We use to sort imports alphabetically.

**Comment by Nachtalb** [comment]
@mbaechtold requested change was made

**Pull request merged**